MzLibUtil/DoubleRange.cs

```diff
@@ -127,5 +127,22 @@ public bool Contains(double item)
         {
             return CompareTo(item).Equals(0);
         }
+
+        public override int GetHashCode()
+        {
+            return Minimum.GetHashCode();
+        }
+
+        public override bool Equals(object obj)
+        {
+            var other = (DoubleRange)obj;
+
+            if(other.Minimum == this.Minimum && other.Maximum == this.Maximum)
```

Comment:
Possibly this:
`return other is DoubleRange && (other as DoubleRange).Minimum.Equals(this.Minimum) && (other as DoubleRange).Maximum.Equals(this.Maximum);`